xeon-phi: ib_core add mic node and scif transport types

The OFED SCIF driver implements a software-emulated RDMA device to allow OFED
based applications, such as Intel MPI, to run on Intel(R) MIC Architecture
without the presence of a physical HCA.  OFED SCIF is only targeted for inter-
node communication within a single platform, where a node is a coprocessor
or the host processor.  This patch adds new node and transport types to the
ib_core kernel module to distinguish this new RDMA interface type.
---
diff -ruN a/drivers/infiniband/core/sysfs.c a1/drivers/infiniband/core/sysfs.c
--- a/drivers/infiniband/core/sysfs.c	2015-09-10 09:29:25.232891731 -0700
+++ a1/drivers/infiniband/core/sysfs.c	2015-09-10 09:31:06.464899773 -0700
@@ -253,6 +253,8 @@
 		return sprintf(buf, "%s\n", "InfiniBand");
 	case IB_LINK_LAYER_ETHERNET:
 		return sprintf(buf, "%s\n", "Ethernet");
+	case IB_LINK_LAYER_SCIF:
+		return sprintf(buf, "%s\n", "SCIF");
 	default:
 		return sprintf(buf, "%s\n", "Unknown");
 	}
@@ -623,6 +625,7 @@
 	case RDMA_NODE_USNIC_UDP: return sprintf(buf, "%d: usNIC UDP\n", dev->node_type);
 	case RDMA_NODE_IB_SWITCH: return sprintf(buf, "%d: switch\n", dev->node_type);
 	case RDMA_NODE_IB_ROUTER: return sprintf(buf, "%d: router\n", dev->node_type);
+	case RDMA_NODE_MIC:	  return sprintf(buf, "%d: MIC\n", dev->node_type);
 	default:		  return sprintf(buf, "%d: <unknown>\n", dev->node_type);
 	}
 }
diff -ruN a/drivers/infiniband/core/verbs.c a1/drivers/infiniband/core/verbs.c
--- a/drivers/infiniband/core/verbs.c	2015-09-10 09:29:25.230891773 -0700
+++ a1/drivers/infiniband/core/verbs.c	2015-09-10 09:31:06.465900979 -0700
@@ -121,6 +121,8 @@
 		return RDMA_TRANSPORT_USNIC;
 	case RDMA_NODE_USNIC_UDP:
 		return RDMA_TRANSPORT_USNIC_UDP;
+	case RDMA_NODE_MIC:
+		return RDMA_TRANSPORT_SCIF;
 	default:
 		BUG();
 		return 0;
@@ -140,6 +142,8 @@
 	case RDMA_TRANSPORT_USNIC:
 	case RDMA_TRANSPORT_USNIC_UDP:
 		return IB_LINK_LAYER_ETHERNET;
+	case RDMA_TRANSPORT_SCIF:
+		return IB_LINK_LAYER_SCIF;
 	default:
 		return IB_LINK_LAYER_UNSPECIFIED;
 	}
diff -ruN a/include/rdma/ib_verbs.h a1/include/rdma/ib_verbs.h
--- a/include/rdma/ib_verbs.h	2015-09-10 09:29:25.240891724 -0700
+++ a1/include/rdma/ib_verbs.h	2015-09-10 09:31:06.466900599 -0700
@@ -75,13 +75,15 @@
 	RDMA_NODE_RNIC,
 	RDMA_NODE_USNIC,
 	RDMA_NODE_USNIC_UDP,
+	RDMA_NODE_MIC,
 };
 
 enum rdma_transport_type {
 	RDMA_TRANSPORT_IB,
 	RDMA_TRANSPORT_IWARP,
 	RDMA_TRANSPORT_USNIC,
-	RDMA_TRANSPORT_USNIC_UDP
+	RDMA_TRANSPORT_USNIC_UDP,
+	RDMA_TRANSPORT_SCIF,
 };
 
 __attribute_const__ enum rdma_transport_type
@@ -91,6 +93,7 @@
 	IB_LINK_LAYER_UNSPECIFIED,
 	IB_LINK_LAYER_INFINIBAND,
 	IB_LINK_LAYER_ETHERNET,
+	IB_LINK_LAYER_SCIF
 };
 
 enum ib_device_cap_flags {
